karapace/schema_registry_apis.py

```diff
@@ -1104,14 +1104,14 @@ async def subjects_schema_post(
                 schema_type=schema_type, schema_str=schema_str, references=references, dependencies=new_schema_dependencies
             )
         except InvalidSchema:
-            self.log.exception("No proper parser found")
+            self.log.warning("Invalid schema: %r", schema_str, exc_info=True)
             self.r(
                 body={
-                    "error_code": SchemaErrorCodes.HTTP_INTERNAL_SERVER_ERROR.value,
+                    "error_code": SchemaErrorCodes.INVALID_SCHEMA.value,
                     "message": f"Error while looking up schema under subject {subject}",
                 },
                 content_type=content_type,
-                status=HTTPStatus.INTERNAL_SERVER_ERROR,
+                status=HTTPStatus.UNPROCESSABLE_ENTITY,
```

Comment:
suggestion: I think we should use HTTP 400 here. It's a non-transient error that clearly points at client error. Unprocessable entity is nice to reserve for situations where "request is valid, but with current server state, it cannot be processed". With an invalid schema, it's not really dependent on server state.

Reply:
This is aligned with Confluent Schema Registry and it does return `422` on `RestInvalidSchemaException`. I'd advice not changing this.

Reply:
Got it TIL 👍